Change to.dtype_layout to c10-full

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#41169 Change to.dtype_layout to c10-full**

-

Differential Revision: [D22289257](https://our.internmc.facebook.com/intern/diff/D22289257/)

smessmer committed:
Change to.dtype_layout to c10-full

-

Differential Revision: [D22289257](https://our.internmc.facebook.com/intern/diff/D22289257/)

[ghstack-poisoned]

smessmer committed:
Update on "Change to.dtype_layout to c10-full"


-

Differential Revision: [D22289257](https://our.internmc.facebook.com/intern/diff/D22289257/)

[ghstack-poisoned]

ezyang commented on aten/src/ATen/test/basic.cpp:
```diff
@@ -361,24 +361,6 @@ TEST(BasicTest, FactoryMethodsTest) {
-  // This is a bug. Requires_grad was set to TRUE but this is being ignored.
-  // Issue https://github.com/pytorch/pytorch/issues/30405
-  ASSERT_FALSE(tensor0.requires_grad());
-  ASSERT_FALSE(tensor0.is_pinned());
```
My recommendation is we fix #30405 first and then revisit this patch.

smessmer replied:
I changed it to keep the requires_grad(false) test as is and changed the requires_grad(true) test to an EXPECT_ANY_THROW.

ezyang commented on aten/src/ATen/core/op_registration/hacky_wrapper_for_legacy_signatures.h:
```diff
@@ -15,6 +15,9 @@ namespace c10 {
 inline c10::optional<MemoryFormat> process_memory_format(const TensorOptions& options, c10::optional<MemoryFormat> memory_format) {
+    TORCH_CHECK(options.requires_grad_opt() == c10::nullopt,
+        "Operators taking TensorOptions cannot take a TensorOptions with "
+        "options.requires_grad set as ", options.requires_grad());
```
It's weird to do a requires grad opt test in `process_memory_format`

smessmer replied:
True. It's unfortunately the only place where I can add it without big codegen changes. Open to ideas.

ezyang replied:
Rename `process_memory_format` then?

smessmer committed:
Update on "Change to.dtype_layout to c10-full"


-

Differential Revision: [D22289257](https://our.internmc.facebook.com/intern/diff/D22289257/)

[ghstack-poisoned]

smessmer committed:
Update on "Change to.dtype_layout to c10-full"


-

Differential Revision: [D22289257](https://our.internmc.facebook.com/intern/diff/D22289257/)

[ghstack-poisoned]

smessmer committed:
Update on "Change to.dtype_layout to c10-full"


-

Differential Revision: [D22289257](https://our.internmc.facebook.com/intern/diff/D22289257/)

[ghstack-poisoned]
